File: DSAlgorithms/Leetcode/ts/src/arrays_hash/2405-optimal_partition_string.ts
const partitionString = (s: string): number =>{
    const current_set = new Set<string>();
    let result = 1;
    for(let i = 0; i < s.length; i++)
    {
        if(current_set.has(s[i]))
        {
            result++;
            current_set.clear();
        }
        current_set.add(s[i]);
    }
    return result;
};

File: DSAlgorithms/Leetcode/arrays/arrays_hash/2405-optimal_partition_string.cpp
class Solution {
public:
    int partitionString(string s) {
       int result = 1; 
       std::unordered_set<char> a_set;
       for(int i = 0; i < s.size(); i++)
       {
           std::unordered_set<char>::iterator it = a_set.find(s[i]);
           if(it != a_set.end())
            {
                result++;
                a_set.clear();
            }
            a_set.insert(s[i]);
       }
       return result;
    }
};